tests/gold_tests/pluginTest/tsapi/tsapi.test.py

```diff
@@ -71,7 +71,15 @@
 tr.Processes.Default.StartBefore(Test.Processes.ts)
 #
 tr.Processes.Default.Command = (
-    'curl --verbose --ipv4 --header "Host: mYhOsT.teSt:{0}" hTtP://loCalhOst:{1}/'.format(server.Variables.Port, ts.Variables.port)
+    r'printf "GET / HTTP/1.1\r\nHost: mYhOsT.teSt:{0}\r\n\r\n" | nc localhost {1}'.format(
```

Comment:
Why nc?

Reply:
To me it's more clear what the exact contents of the HTTP message is when I use nc rather than curl.

Reply:
I prefer consistently using curl for all HTTP versions as long as curl can do the same thing, because I can clearly see the tests do the same things on different versions.
If you think we all should use printf and nc for H1 test cases, you should make another PR to make the change for all H1 tests, and let's see what other people think. I don't think this change has to be done on this PR.

Reply:
@bryancall what is your thinking about this? @d2r ?

I have one test run with the scheme and host in the URL, and one with just the path in the URL (with the host in a host header and no explicit scheme).  Have you seen how long the manpage for curl is?  Does it even say whether or not the scheme/host are in the request URL?  If I have to revisit this test case at some point in the future, I'd rather not have to have to dig into all the quirks of curl to understand it, given that there is a more straight-forward alternative.

Reply:
I'd say "Have you seen how long the H2 spec is?". I would definitely not make a binary data and send it with "openssl s_client" for this test.

Reply:
How do I control whether or not curl puts the scheme and host in the HTTP Request URL?

Reply:
`curl -x http://yourats/ http://foo/` puts them in RequestURL.

Reply:
What do you mean? -x is the abbreviation of --proxy.  What does that have to do with controlling whether or not the scheme and host are in the URL in the HTTP request?

Reply:
https://tools.ietf.org/html/rfc7230#section-5.3.2
> When making a request to a proxy, other than a CONNECT or server-wide
   OPTIONS request (as detailed below), a client MUST send the target
   URI in absolute-form as the request-target.

Reply:
I believe that, under the terminology of the RFC, the term "proxy" only applies to forward proxies.  A reverse proxy is called a "gateway".  I don't think the restriction you reference applies to gateways.

Reply:
If you use --proxy, curl thinks the server is a forward proxy regardless of your server configuration, and it makes the RequestURL you need.

Reply:
OK I think I see what you mean now:
```
$ curl --verbose http://httpbin.org/ip 2>&1 | grep '^> '
> GET /ip HTTP/1.1
> Host: httpbin.org
> User-Agent: curl/7.66.0-DEV
> Accept: */*
> 
$ curl --verbose -x httpbin.org:80 http://httpbin.org/ip 2>&1 | grep '^> '
> GET http://httpbin.org/ip HTTP/1.1
> Host: httpbin.org
> User-Agent: curl/7.66.0-DEV
> Accept: */*
> Proxy-Connection: Keep-Alive
> 
$
```
But to me this makes the test harder rather than easier to understand and maintain.  Notice that curl inserts several headers that are not relevant to the test.

Reply:
> Notice that curl inserts several headers that are not relevant to the test.

I think they are relevant because they may change ATS's behavior. I guess major clients send Proxy-Client when they use absolute form, so testing the behavior without the header may be missing the point. And we can easily make natural requests by using curl. I don't think I can always do it manually.